pkg/cmd/cli/plugin/add.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	ark "github.com/heptio/ark/pkg/apis/ark/v1"
+	"github.com/heptio/ark/pkg/client"
+	"github.com/heptio/ark/pkg/cmd"
+)
+
+func NewAddCommand() *cobra.Command {
+	var kubeconfig string
+
+	c := &cobra.Command{
+		Use:   "add NAME IMAGE",
+		Short: "Add a plugin",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 2 {
+				cmd.CheckError(errors.New("you must specify exactly two arguments, the plugin container's name and image"))
+			}
+
+			clientConfig, err := client.Config(kubeconfig, fmt.Sprintf("%s-%s", c.Parent().Name(), c.Name()))
+			if err != nil {
+				cmd.CheckError(err)
+			}
+
+			kubeClient, err := kubernetes.NewForConfig(clientConfig)
+			if err != nil {
+				cmd.CheckError(err)
+			}
+
+			arkDeploy, err := kubeClient.Apps().Deployments(ark.DefaultNamespace).Get("ark", metav1.GetOptions{})
+			if err != nil {
+				cmd.CheckError(err)
+			}
+
+			plugin := v1.Container{
+				Name:            args[0],
```

Comment:
Maybe we could derive `Name` from the image, so the command is just `ark plugin add IMAGE`? Or if deriving the name is annoying, we could just call it plugin-0001, plugin-0002, etc.